scripts/count_gene_mutations.R

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env Rscript
+
+# Count the number of samples mutated for each gene in a MAF file.
+
+# This script reads in a MAF file and writes out a table of the genes with
+# mutations. The table includes the number of samples that have at least one 
+# mutation in the gene, as well as the total number of mutations detected across
+# all samples.
+
+# Option descriptions:
+#
+# --maf :  File path to MAF file to be analyzed. Can be .gz compressed.
+# --outfile : The name of the output file to create
+# --vaf: Minimum variant allele fraction of mutations to include.
+# --min_depth: Minimum sequencing depth to call mutations.
+# --include_syn: Flag to include synonymous mutations in counts.
+
+# Example invocation:
+#
+# Rscript count_gene_mutations.R \
+#   --maf mutations.maf.tsv.gz \
+#   --outfile gene_counts.tsv
+
+# Load libraries
+library(optparse)
+library(magrittr)
+
+# Set up options
+option_list <- list(
+  make_option(
+    opt_str = c("--maf", "-m"),
+    type = "character",
+    default = NA,
+    help = "File path of MAF file to be analyzed. Can be .gz compressed."
+  ),
+  make_option(
+    opt_str = c("--outfile", "-o"),
+    type = "character",
+    default = "gene_counts.tsv",
+    help = "File path where output table will be placed."
+  ),
+  make_option(
+    opt_str = c("--vaf", "-v"),
+    type = "numeric",
+    default = 0.05,
+    help = "Minimum variant allele fraction to include (default 0.05)",
+    metavar = "numeric"
+  ),
+  make_option(
+    opt_str = c("--min_depth", "-d"),
+    type = "numeric",
+    default = 0,
+    help = "Minimum sequencing depth to include (default 0)",
+    metavar = "numeric"
+  ),
+  make_option(
+    opt_str = "--include_syn",
+    action = "store_true",
+    default = FALSE,
+    help = "Include synonymous coding mutations"
+  )
+)
+# Parse options
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# Check input files
+if(!file.exists(opts$maf)){
+  stop("The specified MAF file does not exist.")
+}
+
+if(!is.null(opts$exclude_genes) && file.exists(opts$exclude_genes)){
+  stop("The specified 'excludes_genes' file does not exist.")
+}
+
+# Read input MAF file
+maf_df <- readr::read_tsv(opts$maf)
+
+# Get the excluded genes list if provided
+if(!is.null(opts$exclude_genes)){
+  exclude_genes <- readr::read_tsv(opts$exclude_file) %>%
+    pull("gene")
+} else {
+  exclude_genes <- c()
+}
+
+# Define the MAF `Consequence` values we are interested in and their classification
+#  based on definitions in http://asia.ensembl.org/Help/Glossary?id=535
+syn_class <- c(
```

Comment:
This is a good teachable moment - spreading code out onto multiple lines makes it easier to read, write, reuse, etc.